Doc/library/urllib.request.rst

```diff
@@ -1557,9 +1549,38 @@ some point in the future.
    :synopsis: Response classes used by urllib.
 
 The :mod:`urllib.response` module defines functions and classes which define a
-minimal file like interface, including ``read()`` and ``readline()``. The
-typical response object is an addinfourl instance, which defines an ``info()``
-method and that returns headers and a ``geturl()`` method that returns the url.
-Functions defined by this module are used internally by the
-:mod:`urllib.request` module.
+minimal file-like interface, including ``read()`` and ``readline()``.
+Functions defined by this module are used internally by the :mod:`urllib.request` module.
+The typical response object is a :class:`urllib.response.addinfourl` instance:
+
+.. class:: urllib.response.addinfourl
+
+.. attribute:: addinfourl.url
+
+   URL of the resource retrieved, commonly used to determine if a redirect was followed.
+
+.. attribute:: addinfourl.headers
+
+   Returns the meta-information of the page, such as headers,
+   in the form of an :func:`email.message_from_string` instance (see
+   `Quick Reference to HTTP Headers <http://jkorpela.fi/http.html>`_)
+
+.. attribute:: addinfourl.code
+
+   Status code returned by server.
+
+.. attribute:: addinfourl.status
+
+   Status code returned by server. Recommended to use :attr:`~addinfourl.status` instead.
+
+.. method:: addinfourl.geturl()
+
+   Returns the URL of the resource retrieved. Recommended to use :attr:`~addinfourl.status` instead.
+
+.. method:: addinfourl.info()
+
+   Returns the response headers. Recommended to use :attr:`~addinfourl.status` instead.
+
+.. method:: addinfourl.getstatus()
 
+   Returns the status. Recommended to use :attr:`~addinfourl.status` instead.
```

Comment:
Should status or code be the obvious recommended attribute?

Reply:
Because `HTTPResponse.status` already exists.

Since originally both `HTTPResponse` and `getaddrurl` had the same three functions, `info()`, `geturl()`, and `getcode()`, and now we're replacing them with attributes instead. Both classes have `.headers` and `.url` so we're good with those two. However, there is no uniform attribute for the status code; we have `HTTPResponse.status` and `getaddrurl.code`.

Since it seems like `HTTPResponse.status` is [already documented](https://docs.python.org/3.7/library/http.client.html#http.client.HTTPResponse.status), I thought it would be better just to rename `getaddrurl.code` to `getaddrurl.status` for consistency's sake.

Reply:
Agreed that following HTTPResponse is best.

Suggested doc for status property: `Returns the status code.`

Doc for code: `Same as status`

Reply:
Sorry I am confused: does addinfourl.code already exist in a released version?  If yes, I think we should keep it, recommend `status`, and not add `code` to HTTPResponse.  If no, let’s keep only getcode (old) and status (recommended, same name as other class).  Sounds good?

Reply:
Oops, I made a typo. What I meant to say is that: `HTTPResponse.status` and `addinfourl.code` already both exist in a released version.

So my solution was to add a property called `addinfourl.status` which would be recommended to use over `addinfourl.code`.